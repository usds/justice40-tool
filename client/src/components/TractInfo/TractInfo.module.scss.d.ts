declare namespace TractInfoNamespace {
    export interface ITractInfoScss {
      tractInfoContainer: string;
      tractInfoLabel: string;
      tractInfoText: string;
    }
  }

declare const TractInfoScssModule: TractInfoNamespace.ITractInfoScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: TractInfoNamespace.ITractInfoScss;
  };

  export = TractInfoScssModule;
