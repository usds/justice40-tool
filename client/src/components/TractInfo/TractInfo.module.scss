@use "../../styles/design-system.scss" as *;
@import "../utils.scss";


.tractInfoContainer {
  display: flex;
  flex-direction: column;
  list-style: none;
  margin: 0;
  padding: 1.2rem 1rem 0 1.2rem;
  font-size: medium;

  .tractInfoLabel {
    @include sidePanelLabelStyle;
  }

}
