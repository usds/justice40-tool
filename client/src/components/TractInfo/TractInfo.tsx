import React from 'react';
import {useIntl} from 'gatsby-plugin-intl';

import * as styles from './TractInfo.module.scss';
import * as EXPLORE_COPY from '../../data/copy/explore';
import * as constants from '../../data/constants';

export interface ITractInfoProps {
  blockGroup: string,
  countyName: string,
  stateName: string,
  population: number,
  sidePanelState: string,
}

const TractInfo = ({blockGroup, countyName, stateName, population, sidePanelState}: ITractInfoProps) => {
  const intl = useIntl();

  return (
    <ul className={styles.tractInfoContainer}>
      <li>
        <span className={styles.tractInfoLabel}>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_CBG_INFO.CENSUS_INFO_TITLE)}
        </span>
      </li>
      <li>
        <span>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_CBG_INFO.CENSUS_BLOCK_GROUP)}
        </span>
        <span>{` ${blockGroup}`}</span>
      </li>
      <li>
        <span>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_CBG_INFO.COUNTY)}
        </span>
        <span>{` ${countyName}`}</span>
      </li>
      <li>
        <span>
          {sidePanelState !== constants.SIDE_PANEL_STATE_VALUES.NATION ?
              intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_CBG_INFO.TERRITORY) :
              intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_CBG_INFO.STATE)
          }
        </span>
        <span>{` ${stateName}`}</span>
      </li>
      <li>
        <span>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_CBG_INFO.POPULATION)}
        </span>
        <span>{` ${population.toLocaleString()}`}</span>
      </li>
    </ul>
  );
};

export default TractInfo;
