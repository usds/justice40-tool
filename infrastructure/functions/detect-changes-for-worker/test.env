REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1