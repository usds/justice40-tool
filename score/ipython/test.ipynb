{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
