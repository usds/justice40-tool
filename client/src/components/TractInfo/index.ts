import TractInfo from './TractInfo';
export default TractInfo;
