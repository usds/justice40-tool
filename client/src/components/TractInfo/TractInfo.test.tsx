import React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import TractInfo from './TractInfo';

describe('rendering of TractInfo Component', () => {
  it('checks if tract info renders correctly for national tracts', () => {
    const blockGroup = '21183920100';
    const countyName = 'Ohio County';
    const stateName = 'Kentucky';
    const population = 3103;
    const sidePanelState = 'Nation';
    const {asFragment} = render(
        <LocalizedComponent>
          <TractInfo
            blockGroup={blockGroup}
            countyName ={countyName}
            stateName ={stateName}
            population ={population}
            sidePanelState ={sidePanelState}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('checks if tract info renders correctly for non-national tracts', () => {
    const blockGroup = '72107954901';
    const countyName = 'Orocovis Municipio';
    const stateName = 'Puerto Rico';
    const population = 3103;
    const sidePanelState = 'Puerto Rico';
    const {asFragment} = render(
        <LocalizedComponent>
          <TractInfo
            blockGroup={blockGroup}
            countyName ={countyName}
            stateName ={stateName}
            population ={population}
            sidePanelState ={sidePanelState}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
