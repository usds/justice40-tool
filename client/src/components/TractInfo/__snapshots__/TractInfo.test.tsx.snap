// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of TractInfo Component checks if tract info renders correctly for national tracts 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <span>
        Tract information
      </span>
    </li>
    <li>
      <span>
        Number:
      </span>
      <span>
         21183920100
      </span>
    </li>
    <li>
      <span>
        County:
      </span>
      <span>
         Ohio County
      </span>
    </li>
    <li>
      <span>
        State: 
      </span>
      <span>
         Kentucky
      </span>
    </li>
    <li>
      <span>
        Population:
      </span>
      <span>
         3,103
      </span>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`rendering of TractInfo Component checks if tract info renders correctly for non-national tracts 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <span>
        Tract information
      </span>
    </li>
    <li>
      <span>
        Number:
      </span>
      <span>
         72107954901
      </span>
    </li>
    <li>
      <span>
        County:
      </span>
      <span>
         Orocovis Municipio
      </span>
    </li>
    <li>
      <span>
        Territory: 
      </span>
      <span>
         Puerto Rico
      </span>
    </li>
    <li>
      <span>
        Population:
      </span>
      <span>
         3,103
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
